### PR TITLE
New blog post: "On Computed Properties vs. Helpers"

### PR DESCRIPTION
**Pull request by marcoow**
@kellyselden: would be great to hear your thoughts as well before releasing this!

### CODE REVIEW

**Comment by kellyselden** [comment]
When viewing https://github.com/simplabs/simplabs.github.io/blob/helpers-vs-computed-properties/blog/_posts/2017-03-21-on-computed-properties-vs-helpers.md, the markdown code escaping gets out of whack throughout the post. Is that expected with Jekyll?

**Comment by marcoow** [comment]
Yep, all renders fine with Jekyll actually.

**Review by Turbo87 — APPROVED** [review]
LGTM 👍

File: blog/_posts/2017-03-21-on-computed-properties-vs-helpers.md [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+```hbs {% raw %}
+{{userToDisplay.name}}
+{% endraw %}```
+
+Comparing this to a solution that moves all of the logic for selecting the user
+into the template via helpers illustrates how much harder it becomes to
+understand the template now:
+
+```hbs {% raw %}
+{{#each (take 1 (filter-by (shuffle users) 'state' 'active')) as |user|}}
```
Comment (kellyselden) [inline blog/_posts/2017-03-21-on-computed-properties-vs-helpers.md:130]: I would do the filter-by first, then the shuffle to match the above order.

File: blog/_posts/2017-03-21-on-computed-properties-vs-helpers.md [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+user's `age` property changes. While arguments that are passed to helpers are
+observed by Ember automatically and the __helper will be re-executed when any
+of these arguments change to a different value, the same is not true when any
+property on any of the arguments changes__. In this case changes of the user's
+`age` property would go unnoticed and the DOM would get out of sync with the
+underlying data.
+
+##### Performance
+
+In general, __performance should be more or less on par for computed properties
```
Comment (kellyselden) [inline blog/_posts/2017-03-21-on-computed-properties-vs-helpers.md:201]: I've heard that this used to be true, until glimmer 2. Now templates are more performant. More research is needed to validate these claims. Perhaps a once-over on CP implementation can yield a similar performance boost.

File: blog/_posts/2017-03-21-on-computed-properties-vs-helpers.md (hunk 1 above)
Comment (marcoow) [reply blog/_posts/2017-03-21-on-computed-properties-vs-helpers.md:130]: Good point - will change

File: blog/_posts/2017-03-21-on-computed-properties-vs-helpers.md (hunk 2 above)
Comment (marcoow) [reply blog/_posts/2017-03-21-on-computed-properties-vs-helpers.md:201]: I'm not sure Glimmer 2 does actually make a difference here. The code that's being executed is pretty much the same both for the CP and the helper solution with the difference that with helpers all arguments will be eagerly evaluated so they might even be slower. I think the difference that Glimmer 2 makes is mostly how it renders the value that it gets from either the CP or the helper and how it makes sure updates are done in a performant way.

**Pull request merged**